scripts/capture_vllm_native_profile.py

```
@@ -4,19 +4,678 @@
+    """Resolve host:port in a subprocess, bounded by the absolute deadline.
+
+    A subprocess is truly killable: if the deadline expires, we terminate
+    it and raise ProfilerTimeoutError.  No thread or resource is retained.
+    """
+    remaining = max(deadline.remaining(), 0.001)
+    code = (
+        "import json,socket,sys;"
+        "rows=socket.getaddrinfo(sys.argv[1],int(sys.argv[2]),0,socket.SOCK_STREAM);"
+        "json.dump([[int(f),int(t),p,c,list(a)] for f,t,p,c,a in rows],sys.stdout)"
+    )
+    proc = subprocess.Popen(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)